src/Common/src/System/Net/Logging/Logging.cs

```diff
@@ -0,0 +1,724 @@
+                }
+                return s_InternalSyncObject;
+            }
+        }
+
+        internal static bool On
+        {
+            get
+            {
+                if (!s_LoggingInitialized)
+                {
+                    InitializeLogging();
+                }
+                return s_LoggingEnabled;
+            }
+        }
+
+        internal static bool IsVerbose(TraceSource traceSource)
+        {
+            return ValidateSettings(traceSource, TraceEventType.Verbose);
+        }
+
+        internal static TraceSource Web
+        {
+            get
+            {
+                if (!s_LoggingInitialized)
+                {
+                    InitializeLogging();
+                }
+                if (!s_LoggingEnabled)
+                {
+                    return null;
+                }
+                return s_WebTraceSource;
+            }
+        }
+
+        internal static TraceSource Http
+        {
+            get
+            {
+                if (!s_LoggingInitialized)
+                {
+                    InitializeLogging();
+                }
+                if (!s_LoggingEnabled)
+                {
+                    return null;
+                }
+                return s_TraceSourceHttpName;
+            }
+        }
+
+        internal static TraceSource HttpListener
+        {
+            get
+            {
+                if (!s_LoggingInitialized)
+                {
+                    InitializeLogging();
+                }
+                if (!s_LoggingEnabled)
+                {
+                    return null;
+                }
+                return s_HttpListenerTraceSource;
+            }
+        }
+
+        internal static TraceSource Sockets
+        {
+            get
+            {
+                if (!s_LoggingInitialized)
+                {
+                    InitializeLogging();
+                }
+                if (!s_LoggingEnabled)
+                {
+                    return null;
+                }
+                return s_SocketsTraceSource;
+            }
+        }
+
+        internal static TraceSource RequestCache
+        {
+            get
+            {
+                if (!s_LoggingInitialized)
+                {
+                    InitializeLogging();
+                }
+                if (!s_LoggingEnabled)
+                {
+                    return null;
+                }
+                return s_CacheTraceSource;
+            }
+        }
+
+        internal static TraceSource WebSockets
+        {
+            get
+            {
+                if (!s_LoggingInitialized)
+                {
+                    InitializeLogging();
+                }
+                if (!s_LoggingEnabled)
+                {
+                    return null;
+                }
+                return s_WebSocketsTraceSource;
+            }
+        }
```

Comment:
Would it make sense to introduce a helper for the boilerplate used in all of these TraceSource-returning properties? e.g.
```C#
private static TraceSource GetTraceSource(ref TraceSource source)
{
    if (!s_LoggingInitialized)
    {
        InitializeLogging();
    }
    return s_LoggingEnabled ? source : null;
}
```
which could then be used like:
```
internal static TraceSource WebSockets { get { return GetTraceSource(ref s_WebSocketsTraceSource); } }
```
or are there concerns around inlining or something like that?